src/main/java/org/apache/maven/shared/dependency/graph/DependencyCollectorRequest.java

```diff
@@ -0,0 +1,162 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.apache.maven.artifact.resolver.filter.ArtifactFilter;
+import org.apache.maven.project.ProjectBuildingRequest;
+import org.apache.maven.shared.dependency.graph.internal.DirectScopeDependencySelector;
+import org.apache.maven.shared.dependency.graph.internal.VerboseJavaScopeSelector;
+import org.eclipse.aether.collection.DependencyGraphTransformer;
+import org.eclipse.aether.collection.DependencySelector;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.graph.manager.DependencyManagerUtils;
+import org.eclipse.aether.util.graph.selector.AndDependencySelector;
+import org.eclipse.aether.util.graph.selector.ExclusionDependencySelector;
+import org.eclipse.aether.util.graph.selector.OptionalDependencySelector;
+import org.eclipse.aether.util.graph.transformer.ConflictResolver;
+import org.eclipse.aether.util.graph.transformer.JavaScopeDeriver;
+import org.eclipse.aether.util.graph.transformer.NearestVersionSelector;
+import org.eclipse.aether.util.graph.transformer.SimpleOptionalitySelector;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * <div>
+ * This class will carry various options used by
+ * {@link DependencyCollectorBuilder#collectDependencyGraph(DependencyCollectorRequest)}
+ * </div>
+ * <div>
+ * There is a set of default values such:
+ * </div>
+ * <div>
+ * DependencySelector
+ * <pre>
+ *  new AndDependencySelector(
+ *             new DirectScopeDependencySelector( JavaScopes.TEST ),
+ *             new DirectScopeDependencySelector( JavaScopes.PROVIDED ),
+ *             new OptionalDependencySelector(),
+ *             new ExclusionDependencySelector() );
+ * </pre>
+ * </div>
+ * <div>
+ * DependencyGraphTransformer
+ * <pre>
+ * new ConflictResolver(
+ *             new NearestVersionSelector(),
+ *             new VerboseJavaScopeSelector(),
+ *             new SimpleOptionalitySelector(),
+ *             new JavaScopeDeriver() );
+ * </pre>
+ * </div>
+ * <div>
+ * configProperties have 2 default values
+ * <pre>
+ *   ConflictResolver.CONFIG_PROP_VERBOSE, true
+ *   DependencyManagerUtils.CONFIG_PROP_VERBOSE, true
+ * </pre>
+ *
+ * </div>
+ * @since 3.2.1
+ */
+public class DependencyCollectorRequest
+{
+
+    private final ProjectBuildingRequest buildingRequest;
+
+    private ArtifactFilter filter;
+
+    private Map<String, Object> configProperties = new HashMap<>();
+
+    private DependencySelector dependencySelector = new AndDependencySelector(
+            new DirectScopeDependencySelector( JavaScopes.TEST ),
+            new DirectScopeDependencySelector( JavaScopes.PROVIDED ),
+            new OptionalDependencySelector(),
+            new ExclusionDependencySelector() );
```

Comment:
well TBH this empty ExclusionDependencySelector doesn't have any effect. we could simply remove it and the code not will not pass tru to do nothing :)